var/spack/repos/builtin/packages/palace/package.py

```diff
@@ -94,8 +98,19 @@ class Palace(CMakePackage):
         depends_on("arpack-ng+shared", when="+shared")
         depends_on("arpack-ng~shared", when="~shared")
 
-    # Palace always builds its own internal MFEM, GSLIB
+    with when("+libxsmm"):
+        depends_on("libxsmm@main")
+        depends_on("libxsmm+shared", when="+shared")
+        depends_on("libxsmm~shared", when="~shared")
+
+    with when("+magma"):
+        depends_on("magma")
+        depends_on("magma+shared", when="+shared")
+        depends_on("magma~shared", when="~shared")
+
+    # Palace always builds its own internal MFEM, libCEED, and GSLIB
     conflicts("mfem")
+    conflicts("libceed")
     conflicts("gslib")
```

Comment:
What is the purpose of this (and even the above `conflict` directives)?  If meant to check if variants are enabled, then they would start with `+` (e.g., for the `gslib` variant it would be `conflicts("+gslib")`).  At this point I'm pretty sure they're adding checks that will never be true.

Reply:
I am not certain of the correct way to do this, but here is what I'm trying to do: Palace builds it's own versions of these packages, and I want to avoid the user installing a dependency of Palace (for example, PETSc), which has one of these packages as a dependency and thus leads to a conflict. Is there a better way to do so? Previously I had `conflicts("^gslib")`, for example, but this was changed by https://github.com/spack/spack/pull/38447. I believe other packages use similar `conflicts` statements for this type of thing (for example, Scotch when it provides METIS).

Reply:
That makes sense.  I'm not familiar with this use case.

Reply:
> That makes sense. I'm not familiar with this use case for the directive.